### PR TITLE
feature/login

### PR DESCRIPTION
**Pull request by andersonmalheiro**

### CODE REVIEW

File: frontend/src/api/error-handler.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { AxiosError } from "axios";
+import { toast } from "react-toastify";
+
+export const errorHandler = (error: AxiosError) => {
```
Comment (edimossilva) [inline frontend/src/api/error-handler.ts:4]: lets check if I understand it properly
1) first `error` is an instance of `AxiosError`
2) later `error` is resigned as something from  `error.response.data.error` ?

if steps 1 and 2 are true this is not good

File: frontend/src/api/error-handler.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  const { response } = error;
+  if (response && response.data) {
+    const { error } = response.data;
+    if (error) {
+      toast(error, {
```
Comment (edimossilva) [inline frontend/src/api/error-handler.ts:9]: the 2 toasts are almost the same, extract?

File: frontend/src/api/services/auth.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    this.httpClient = httpClient;
+  }
+
+  public async authenticate(user: UserLogin): Promise<User | undefined> {
+    try {
+      const response = await this.httpClient.post<User>(`/users/sign_in`, {
+        user,
+      });
+
+      if (response && response.data) {
```
Comment (edimossilva) [inline frontend/src/api/services/auth.ts:16]: if this condition returns false what happens?

File: frontend/src/components/app-input/app-input.styles.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import styled from "styled-components";
+
+export const InputWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+
+  label {
+    font-weight: bold;
+    margin: 0 0 5px 5px;
```
Comment (edimossilva) [inline frontend/src/components/app-input/app-input.styles.ts:9]: Looks like this 5px should be variable for design consistency. This should be applied in more places

File: frontend/src/pages/login.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+  const handleFormSubmit = (data: any) => {
+    login(data);
+  };
+  const [loading, setLoading] = useState(false);
+
+  const login = async (data: UserLogin) => {
+    try {
+      setLoading(true);
+      const response = await authService.authenticate(data);
+      console.log(`response`, response);
```
Comment (edimossilva) [inline frontend/src/pages/login.tsx:32]: leftover? should lint track this?

**Review by edimossilva — CHANGES_REQUESTED** [review]
I've written something that maybe is relevant but my main request is to add tests.

File: frontend/src/api/error-handler.ts (hunk 1 above)
Comment (andersonmalheiro) [reply frontend/src/api/error-handler.ts:4]: I will rename the variable and improve the code

File: frontend/src/api/error-handler.ts (hunk 2 above)
Comment (andersonmalheiro) [reply frontend/src/api/error-handler.ts:9]: Ok, I will make a function to render the toast

File: frontend/src/api/services/auth.ts (hunk 3 above)
Comment (andersonmalheiro) [reply frontend/src/api/services/auth.ts:16]: Fixed on error-handler refactor

File: frontend/src/pages/login.tsx (hunk 5 above)
Comment (andersonmalheiro) [reply frontend/src/pages/login.tsx:32]: Removed

**Review by edimossilva — APPROVED** [review]
Niceee, approved!!!!!!!!

**Pull request merged**